ProjFS.Mac/PrjFSKext/KauthHandlerTestable.hpp

```diff
@@ -12,5 +13,20 @@
 
 KEXT_STATIC_INLINE bool FileFlagsBitIsSet(uint32_t fileFlags, uint32_t bit);
 KEXT_STATIC_INLINE bool ActionBitIsSet(kauth_action_t action, kauth_action_t mask);
+KEXT_STATIC_INLINE bool TryGetFileIsFlaggedAsInRoot(vnode_t vnode, vfs_context_t context, bool* flaggedInRoot);
```

Comment:
The mock KPIs currently don't use the context pointer for anything, but we could change that to check it gets passed through correctly and consistently. (some APIs silently accept `nullptr` contexts in some circumstances while KPing in others, for example)

So I'd agree, we should probably try to write our tests consistently. Where we don't really care about vfs contexts, we can just create one in `setUp` and destroy it in `tearDown` and use it in each of the test methods. That's not significantly more work than passing `nullptr`.

Reply:
@wilbaker since KauthHandlerTestable is a header file if we add _Nonnull for one paramter we must add it for every parameter.  

KauthHandler.cpp does not have this requirement since it isn't a header.

So to get the signatures to match we'd either need to
1. Add a nonnullable/nullable tag to every param in Kauthhandler.cpp which creates a lot of noise.  Don't do that.
2. Use " #pragma clang assume_nullable begin/end" which @pmj has advised against.

For now I'll complete this PR without adding `_Nonnull `but we can pick the discussion up at a later time.

Reply:
> @wilbaker since KauthHandlerTestable is a header file if we add _Nonnull for one paramter we must add it for every parameter.

> KauthHandler.cpp does not have this requirement since it isn't a header.

Ahh I didn't realize there was a difference in the requirements for these specifiers (between headers and source).

> Add a nonnullable/nullable tag to every param in Kauthhandler.cpp which creates a lot of noise. Don't do that.

I actually think tagging all of the pointer params isn't such a bad idea.  It would get us some additional protection from the compiler and help convey the expectations for those parameters.  

However, I agree there's no need to clean that up now (or block this PR on it).

Reply:
> Ahh I didn't realize there was a difference in the requirements for these specifiers (between headers and source).

This would also explain why I didn't hit this issue with the vnode cache, I believe I specified the nullability for all pointer parameters.